### PR TITLE
Add automatic disk resizing when disk_size is specified

### PR DESCRIPTION
**Pull request by ehelms**
If a box is defined with a larger disk size through the use of
disk_size parameter the user then has to manually resize the filesystem.
With the introduction of CentOS 8, which has a base disk size of
10GB within the image this operation will be prevalent for nearly
all Katello production and development deployments through Vagrant.

This adds automatic disk resizing by running a playbook as a
provisioner if the disk_size parameter is defined.

### CODE REVIEW

File: playbooks/resize_disk.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      package:
+        name: cloud-utils-growpart
+
+    - name: Install e2fsprogs package
+      package:
+        name: e2fsprogs
+      when: ansible_distribution_major_version == '7'
+
+    - name: Extend partition
+      shell: growpart /dev/vda 1
```
Comment (ekohl) [inline playbooks/resize_disk.yaml:15]: Isn't command safer than shell? I thought ansible-lint complained about that

File: playbooks/resize_disk.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        - '"NOCHANGE" not in output.stdout'
+        - output.rc is not 0
+
+    - name: Grow XFS filesystem
+      shell: xfs_growfs -d /
+      when: ansible_distribution_major_version == '8'
+
+    - name: Grow ext4 filesystem
+      shell: resize2fs /dev/vda1
+      when: ansible_distribution_major_version == '7'
```
Comment (ekohl) [inline playbooks/resize_disk.yaml:31]: Isn't it better to detect the partition type and resize it accordingly?

**Review by ekohl — CHANGES_REQUESTED** [review]
This assumes a CentOS only world.

File: playbooks/resize_disk.yaml (hunk 2 above)
Comment (ehelms) [reply playbooks/resize_disk.yaml:31]: I don't know. I could try that.

File: playbooks/resize_disk.yaml (hunk 1 above)
Comment (ehelms) [reply playbooks/resize_disk.yaml:15]: Generally yes, but depends on the use case. I had to go with shell here due to command erroring out. I'd have to dig up the error.

**Comment by ehelms** [comment]
This is now a bit more dynamic for more use cases and has been tested on CentOS 7 and 8, Ubuntu 1604 and Debian 10 images.

File: playbooks/foreman_repos.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
```
Comment (ekohl) [inline playbooks/foreman_repos.yml:1]: This looks unrelated

File: playbooks/resize_disk.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+---
+- hosts: all
+  become: true
+  tasks:
+    - name: Find root partition
+      shell: df -hT | grep /$
```
Comment (ekohl) [inline playbooks/resize_disk.yaml:6]:
```suggestion
      shell: df --output=source,fstype / | tail -n 1
```

File: playbooks/resize_disk.yaml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    - name: Extend partition
+      command: growpart /dev/vda {{ root_partition.stdout.split(' ')[0].replace('/dev/vda', '') }}
+      register: output
+      ignore_errors: true
+
+    - name: Fail if growpart fails, do not fail if NOCHANGE
+      fail:
+      when:
+        - '"NOCHANGE" not in output.stdout'
+        - output.rc != 0
```
Comment (ekohl) [inline playbooks/resize_disk.yaml:33]: Does't this make the `vagrant provision` command fail on runs after the first one?

**Review by ekohl — CHANGES_REQUESTED** [review]

File: playbooks/resize_disk.yaml (hunk 5 above)
Comment (ehelms) [reply playbooks/resize_disk.yaml:33]: Nope. This is an `and` condition. So if it reports a non-zero return code but does not contain NOCHANGE in the output, then the error is likely a real error and we should fail. For whatever reason, NOCHANGE causes a return code of `1` and from all my testing the same is true if oyu attempt it with `--dry-run`.

File: playbooks/resize_disk.yaml [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        name: cloud-guest-utils
+      when: ansible_os_family == 'Debian'
+
+    - name: Install e2fsprogs package
+      package:
+        name: e2fsprogs
+      when: '"ext4" in root_partition.stdout'
+
+    - name: Extend partition
+      command: growpart /dev/vda {{ root_partition.stdout.split(' ')[0].replace('/dev/vda', '') }}
```
Comment (ekohl) [inline playbooks/resize_disk.yaml:25]: I am wondering if this is always `vda`. `vda` is naming for virtio but on virtualbox it may fail if they use `sda`. Not sure if we have those, but worth considering.

**Review by ekohl — COMMENTED** [review]
I'd be OK with trying this.

File: playbooks/resize_disk.yaml (hunk 6 above)
Comment (ehelms) [reply playbooks/resize_disk.yaml:25]: Can cross that bridge if someone gets there.

File: playbooks/resize_disk.yaml [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+      register: output
+      ignore_errors: true
+
+    - name: Fail if growpart fails, do not fail if NOCHANGE
+      fail:
+      when:
+        - '"NOCHANGE" not in output.stdout'
+        - output.rc != 0
```
Comment (evgeni) [inline playbooks/resize_disk.yaml:33]: Instead of using `fail` (without any message) here, wouldn't it be possible to make a more correct [`failed_when` in the `command` step](https://docs.ansible.com/ansible/latest/user_guide/playbooks_error_handling.html#defining-failure) above?

something like:

```yaml
    - name: Extend partition
      command: growpart /dev/vda {{ root_partition.stdout.split(' ')[0].replace('/dev/vda', '') }}
      register: growpart_output
      failed_when:
        - '"NOCHANGE" not in growpart_output.stdout'
        - growpart_output.rc != 0
```

BTW, are the extra quotes around the `… not in …` required?

Comment (ehelms) [reply playbooks/resize_disk.yaml:33]: Yes, the quotes appear to be needed.

Comment (evgeni) [reply playbooks/resize_disk.yaml]: :crying_cat_face:

**Review by evgeni — APPROVED** [review]

**Review by ekohl — APPROVED** [review]
